pkg/sign/manifest.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sign
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	tarball "github.com/cert-manager/release/pkg/release/tar"
+)
+
+// CertManagerManifests takes a path to a cert-manager-manifests.tar.gz file, loads it into
+// memory and signs anything inside the archive which is signable; currently,
+// the helm chart located at "deploy/chart/cert-manager.tgz" is signed, and a
+// signature "deploy/chart/cert-manager.tgz.prov" will be added.
+// The cert-manifests.tar.gz file is changed in-place.
+func CertManagerManifests(ctx context.Context, key string, path string) error {
+	if key == "" {
+		return fmt.Errorf("can't sign manifests without a signing key specified")
+	}
+
+	manifestLocation := "deploy/chart/cert-manager.tgz"
+
+	// 1. Create temp dir for chart archive to be extracted to
+	// (Helm signing requires a filename, not a reader, so we have to write to disk here)
+	tmpDest, err := os.MkdirTemp("", "extracted-manifests-")
```

Comment:
We should namespace our temp dirs with a useful label.

```suggestion
	tmpDest, err := os.MkdirTemp("", "cmrel-extracted-manifests-")
```

Reply:
I guess we have to use a real directory here instead of a virtual filesystem or just memory because of a limitation with helm itself?